telethon/client/chats.py

```diff
@@ -762,4 +762,229 @@ def action(
+        Returns
+            The resulting :tl:`Updates` object.
+
+        Example
+            .. code-block:: python
+
+                # Allowing `user` to pin messages in `chat`
+                client.edit_admin(chat, user, pin_messages=True)
+        """
+        entity = await self.get_input_entity(entity)
+        user = await self.get_input_entity(user)
+        if not isinstance(user, types.InputPeerUser):
+            raise ValueError('You must pass a user entity')
+
+        if isinstance(entity, types.InputPeerChannel):
+            return await self(functions.channels.EditAdminRequest(
+                entity,
+                user,
+                types.ChatAdminRights(
+                    change_info=change_info,
+                    post_messages=post_messages,
+                    edit_messages=edit_messages,
+                    delete_messages=delete_messages,
+                    ban_users=ban_users,
+                    invite_users=invite_users,
+                    pin_messages=pin_messages,
+                    add_admins=add_admins
+                )
+            ))
+        elif isinstance(entity, types.InputPeerChat):
+            return await self(functions.messages.EditChatAdminRequest(entity, user, is_admin=is_admin))
+        else:
+            raise ValueError('You must pass either a channel or a supergroup or a normal group')
+
+    async def edit_permission(
+            self: 'TelegramClient',
+            entity: 'hints.EntityLike',
+            user: 'typing.Optional[hints.EntityLike]' = None,
+            until_date: 'hints.DateLike' = None,
+            *,
+            view_messages: bool = None,
+            send_messages: bool = None,
+            send_media: bool = None,
+            send_stickers: bool = None,
+            send_gifs: bool = None,
+            send_games: bool = None,
+            send_inline: bool = None,
+            send_polls: bool = None,
+            change_info: bool = None,
+            invite_users: bool = None,
+            pin_messages: bool = None) -> types.Updates:
+        """
+        Edits user restrictions in a chat.
+
+        Raises an error if wrong a combination of rights are given
+        (e.g. you don't have enough permissions to revoke one).
+
+        Arguments
+            entity (`entity`):
+                The channel or megagroup where the restriction should happen.
+
+            user (`entity`, optional):
+                If specified, the permission will be changed for the specific user.
+                If left as ``None``, the default chat permissions will be updated.
+
+            until_date (`DateLike`, optional):
+                When the user will be unbanned.
+
+                If the due date or duration is longer than 366 days or shorter than
+                30 seconds, the ban will be forever. Defaults to ``0`` (ban forever).
+
+            view_messages (`bool`, optional):
+                Whether the user is able to view messages or not.
```

Comment:
Are you sure setting this to `False` bans the person? If I remember, it needs to be the opposite for it to have that effect.

Reply:
yeah should I inverse them or leave it like telegram does it?

Reply:
Well, the current documentation is wrong, and if we do it the way Telegram does, it's probably very confusing. So I vote for reversing them.